db/blip_sync_context.go

```diff
@@ -354,10 +354,20 @@ func (bsc *BlipSyncContext) handleChangesResponse(ctx context.Context, sender *b
 				knownRevsByDoc[docID] = knownRevs
 			}
 
-			// The first element of the knownRevsArray returned from CBL is the parent revision to use as deltaSrc
+			// The first element of the knownRevsArray returned from CBL is the parent revision to use as deltaSrc for
+			// revtree clients, for HLV clients the first element is the HLV
 			if bsc.useDeltas && len(knownRevsArray) > 0 {
 				if revID, ok := knownRevsArray[0].(string); ok {
-					deltaSrcRevID = revID
+					if bsc.useHLV() {
+						msgHLV, err := extractHLVFromBlipMessage(revID)
+						if err != nil {
+							base.ErrorfCtx(ctx, "Invalid known rev format for hlv on doc: %s", docID)
```

Comment:
If we can't parse this, should we fall back to sending full body instead of erroring?  Or will it cause problems elsewhere?

Reply:
On a related note, does the loop in line 374 also need to handle HLV format, or does that happen elsewhere?

Reply:
Good spot! Have updated to reflect this, let me know what you think!